RabbitMQ.Stream.Client/RawProducer.cs

```diff
@@ -197,26 +197,9 @@ internal void PreValidateBatch(List<(ulong, Message)> messages)
             }
         }
 
-        private async Task SemaphoreWait()
-        {
-            if (!await _semaphore.WaitAsync(0) && !_client.IsClosed)
-            {
-                // Nope, we have maxed our In-Flight messages, let's asynchronously wait for confirms
-                if (!await _semaphore.WaitAsync(TimeSpan.FromSeconds(1)).ConfigureAwait(false))
-                {
-                    _logger.LogWarning("Semaphore Wait timeout during publishing.");
-                }
-            }
```

Comment:
Remove this Async Await. In some case, it raises `semaphorefullexception` 
after:
```
_logger.LogWarning("Semaphore Wait timeout during publishing.");
```
the code continue to work failing the scope of the `lock`